service-registry/registry-service-center/src/main/java/org/apache/servicecomb/serviceregistry/task/MicroserviceRegisterTask.java

```diff
@@ -86,6 +86,24 @@ protected boolean doRegister() {
     if (!StringUtils.isEmpty(serviceId)) {
       // This microservice has been registered, so we just use the serviceId gotten from service center
       microservice.setServiceId(serviceId);
+
+      // Need to update microservice properties if we have modified or added properties of microservices.
+      Microservice microserviceTemp = new Microservice();
```

Comment:
When you update microservcie propeties, the client may not be notified for this change.  And why you implement this feature and does your scenario need notification?

Reply:
To support the gray deployment function。